src/services/Ildcp.ts

```diff
@@ -0,0 +1,40 @@
+import { IldcpInfo } from '../schemas/IldcpInfo'
+import * as ILDCP from 'ilp-protocol-ildcp'
+import { Injector } from 'reduct'
+import Config from './Config'
+
+import { create as createLogger } from '../common/log'
+const log = createLogger('ILDCP')
+const makePlugin = require('ilp-plugin')
+
+export default class Ildcp {
+  private info: IldcpInfo
+  private config: Config
+
+  constructor (deps: Injector) {
+    this.config = deps(Config)
+  }
+
+  async init () {
+    if (this.config.devMode) {
+      this.info = this.config.devIldcp
+    } else {
+      log.debug('fetching dcp information')
+      const dcpPlugin = makePlugin()
+      await dcpPlugin.connect()
+      this.info = await ILDCP.fetch(dcpPlugin.sendData.bind(dcpPlugin))
```

Comment:
we should clean up the plugin after this

Reply:
Is calling `disconnect` sufficient?
```suggestion
      this.info = await ILDCP.fetch(dcpPlugin.sendData.bind(dcpPlugin))
      await dcpPlugin.disconnect()
```

Reply:
yeah, that should work